runtime/gc_glue_java/CollectorLanguageInterfaceImpl.cpp

```diff
@@ -702,11 +702,27 @@ MM_CollectorLanguageInterfaceImpl::scavenger_switchConcurrentForThread(MM_Enviro
 		}
 		vmThread->evacuateBase = gsBase;
 		vmThread->evacuateTop = gsTop;
+#if defined(J9VM_GC_COMPRESSED_POINTERS)
+		vmThread->evacuateBaseCompressed = _extensions->accessBarrier->convertTokenFromPointer((mm_j9object_t)gsBase);
+		vmThread->evacuateTopCompressed = _extensions->accessBarrier->convertTokenFromPointer((mm_j9object_t)gsTop);
+#endif /* J9VM_GC_COMPRESSED_POINTERS */
 		vmThread->privateFlags |= J9_PRIVATE_FLAGS_CONCURRENT_SCAVENGER_ACTIVE;
 		j9gs_enable(&vmThread->gsParameters, _extensions->getConcurrentScavengerPageStartAddress(), _extensions->getConcurrentScavengerPageSectionSize(), bitMask);
     } else {
 		j9gs_disable(&vmThread->gsParameters);
+		/*
+		 * By setting evacuateTop to NULL and evacuateBase to ~evacuateTop
+		 * it gives an empty range that contains no address. Therefore,
+		 * when decide whether to read barrier, a simple range is sufficient and
+		 * checking J9_PRIVATE_FLAGS_CONCURRENT_SCAVENGER_ACTIVE can be skipped.
+		 */
 		vmThread->privateFlags &= ~J9_PRIVATE_FLAGS_CONCURRENT_SCAVENGER_ACTIVE;
+		vmThread->evacuateTop = NULL;
+		vmThread->evacuateBase = (void*)~(UDATA)vmThread->evacuateTop;
+#ifdef J9VM_GC_COMPRESSED_POINTERS
+		vmThread->evacuateTopCompressed = 0;
+		vmThread->evacuateBaseCompressed = ~vmThread->evacuateTopCompressed;
```

Comment:
I actually prefer using (void *)UDATA_MAX, it's more in coding spirit with the rest of GC code.

(I would actually even more prefer UINT_MAX, but the code is still heavily using UDATA_MAX, so let's be consistent)

Reply:
Updated.